prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -376,6 +384,14 @@ func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, callBack func(), o
 				errs = append(errs, fmt.Errorf("failed to construct manager for cluster %s: %w", name, err))
 				return
 			}
+
+			// Check to see if we are able to perform actions against pods in
+			// the build cluster. The actions are given in requiredTestPodVerbs.
+			if err := checkAuthorizations(config, options.Namespace, requiredTestPodVerbs); err != nil {
+				errs = append(errs, fmt.Errorf("failed pod resource authorization check: %w", err))
+				return
+			}
```

Comment:
We should check the authorizations without holding the lock so that all the build cluster authorization checks can happen in parallel. I think we only need to hold `lock` when updating `res` or `errs`.

Reply:
Nice catch. I just added more explicit locks/unlocks around here (and also for `res` and `errs`).